ddapm_test_agent/vcr_proxy.py

```diff
@@ -200,8 +217,11 @@ async def proxy_request(request: Request, vcr_cassettes_directory: str, vcr_ci_m
         auth = AWS4Auth(aws_access_key, AWS_SECRET_ACCESS_KEY, AWS_REGION, AWS_SERVICES[provider])
         request_kwargs["auth"] = auth
 
-    with get_vcr(provider, vcr_cassettes_directory).use_cassette(cassette_file_name):
-        provider_response = requests.request(**request_kwargs)
+    def _make_request():
+        with get_vcr(provider, vcr_cassettes_directory, vcr_ignore_headers).use_cassette(cassette_file_name):
+            return requests.request(**request_kwargs)
+
+    provider_response = await asyncio.to_thread(_make_request)
```

Comment:
made this change to avoid event-loop blocking issues when testing with the `aiohttp_server` fixture. i _think_ this is safe for having generally (tested against usage in `dd-trace-js` test jobs), but please let me know otherwise! i can use a different fixture to create the dummy server otherwise